src/main/java/use_case/search/SearchInteractor.java

```diff
@@ -0,0 +1,75 @@
+package use_case.search;
+
+import data_access.movies.MovieDBDataAccessException;
+import entity.Movie;
+import use_case.generate_recommendations.MovieDBDataAccessInterface;
+
+/**
+ * Interactor for the search by criteria use case.
+ */
+public class SearchInteractor implements SearchInputBoundary {
+    private final MovieDBDataAccessInterface movieDBDataAccess;
+    private final SearchOutputBoundary outputBoundary;
+
+    public SearchInteractor(MovieDBDataAccessInterface movieDBDataAccess, SearchOutputBoundary outputBoundary) {
+        this.movieDBDataAccess = movieDBDataAccess;
+        this.outputBoundary = outputBoundary;
+    }
+
+    @Override
+    public void execute(SearchByCriteriaInputData inputData) {
+        Movie matchingMovie = null;
+        String movieDetails;
+
+        // Check if the list of keywords is empty or if the first keyword is empty or contains only whitespace
+        if (inputData.getKeywords().isEmpty() || inputData.getKeywords().get(0).trim().isEmpty()) {
+            movieDetails = "Error: Please provide a valid movie title.";
+            outputBoundary.prepareFailView(new SearchByCriteriaOutputData(movieDetails));
+        }
+        else {
+            try {
+                // Validate the category (Movie/TV show)
+                if (validateCategory(inputData.getCategory())) {
+                    final String searchKeyword = inputData.getKeywords().get(0).trim();
+
+                    // Call DAO to search for the movie if a valid keyword is provided
+                    if (!searchKeyword.isEmpty()) {
+                        matchingMovie = movieDBDataAccess.getMovie(searchKeyword);
+                    }
+                }
+
+                // Convert result to output data
+                if (matchingMovie != null) {
```

Comment:
I don't think this clause would ever be triggered, since in the case where TMDB doesn't return a valid movie id for the user input, the getMovie method in the MovieDB DAO would raise an invalid index exception (which should probably be caught and converted to a MovieDB DA Exception). That is to say, I think you can take it out and shrink the "try" clause strictly to the if statement that gets something from the database.